kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/ContentToolFsVolumeFactory.java

```diff
@@ -350,6 +352,11 @@ public long getSize(SakaiFsItem fsi) {
             String id = fsi.getId();
             try {
                 if (contentHostingService.isCollection(id)) {
+                    // In sites with many resources this causes severe performance issues.
+                    boolean showFolderSize = serverConfigurationService.getBoolean("elfinder.calculate.folder.sizes", true);
+                    if (!showFolderSize) {
+                        return 0;
+                    }
                     return contentHostingService.getCollectionSize(id);
```

Comment:
```suggestion
                    return (showFolderSize) ? contentHostingService.getCollectionSize(id) : 0;
```
What do you think of this?